components/proxy/src/main/java/com/hotels/styx/proxy/HttpErrorStatusCauseLogger.java

```diff
@@ -29,7 +30,13 @@
  * Wrapper for {@link HttpErrorStatusListener} that also logs {@link Throwable}s.
  */
 public class HttpErrorStatusCauseLogger implements HttpErrorStatusListener {
+
     private static final Logger LOG = getLogger(HttpErrorStatusCauseLogger.class);
+    private HttpMessageFormatter formatter;
+
+    public HttpErrorStatusCauseLogger(HttpMessageFormatter formatter) {
+        this.formatter = formatter;
+    }
```

Comment:
Formatter could be `final`, and the constructor would need a null check.

Reply:
fixed